core/controllers/editor.py

```diff
@@ -157,28 +155,27 @@ def get(self, exploration_id):
             'DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR': (
                 DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR.value),
             'additional_angular_modules': additional_angular_modules,
-            'can_delete': rights_manager.Actor(
-                self.user_id).can_delete(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_edit': can_edit,
-            'can_modify_roles': rights_manager.Actor(
-                self.user_id).can_modify_roles(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_publicize': rights_manager.Actor(
-                self.user_id).can_publicize(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_publish': rights_manager.Actor(
-                self.user_id).can_publish(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_release_ownership': rights_manager.Actor(
-                self.user_id).can_release_ownership(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_unpublicize': rights_manager.Actor(
-                self.user_id).can_unpublicize(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
-            'can_unpublish': rights_manager.Actor(
-                self.user_id).can_unpublish(
-                    feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id),
+            'can_delete': rights_manager.check_can_delete_exploration(
+                self.user_id, self.actions, exploration_rights),
+            'can_edit': rights_manager.check_can_edit_activity(
+                self.user_id, self.actions, feconf.ACTIVITY_TYPE_EXPLORATION,
+                exploration_rights),
+            'can_modify_roles': (
+                rights_manager.check_can_modify_exploration_roles(
+                    self.user_id, self.actions, exploration_rights)),
+            'can_publicize': rights_manager.check_can_publicize_exploration(
```

Comment:
Just a note for a future PR: one alternative way of handling some of the refactoring I mentioned earlier (where hte routes to the backend were different for editors and moderators) is to get rid of the "can publicize" functionality altogether. We can unpublicize the existing "publicized" explorations later and remove the functionality entirely, but removing the "can publicize" part of things seems like a decent first step. Probably better to leave it for a separate PR though.